watch/source.go

```diff
@@ -53,20 +58,33 @@ func NewSource(poll SourcePollFn, logger xlog.Logger) Source {
 type source struct {
 	sync.RWMutex
 
-	poll   SourcePollFn
-	w      Watchable
-	logger xlog.Logger
-	closed bool
+	poll        SourcePollFn
```

Comment:
Just say this is a function pointer - we should really make the poller be a interface, so it could hold state about the last poll.